src/subscript/co2containment/calculate.py

```diff
@@ -0,0 +1,218 @@
+                ),
+                ContainedCo2(
+                    w.date, sum(w.aqu_phase[is_contained]), "aqueous", "contained"
+                ),
+                ContainedCo2(
+                    w.date, sum(w.aqu_phase[is_outside]), "aqueous", "outside"
+                ),
+                ContainedCo2(
+                    w.date, sum(w.aqu_phase[is_hazardous]), "aqueous", "hazardous"
+                ),
+            ]
+        ]
+    zone_map = {z: co2_data.zone == z for z in np.unique(co2_data.zone)}
+    if calc_type == CalculationType.VOLUME_EXTENT:
+        return [
+            c
+            for w in co2_data.data_list
+            for zn, zm in zone_map.items()
+            for c in [
+                ContainedCo2(
+                    w.date,
+                    sum(w.volume_coverage[is_contained & zm]),
+                    "gas",
+                    "contained",
+                    zn,
+                ),
+                ContainedCo2(
+                    w.date,
+                    sum(w.volume_coverage[is_outside & zm]),
+                    "gas",
+                    "outside",
+                    zn,
+                ),
+                ContainedCo2(
+                    w.date,
+                    sum(w.volume_coverage[is_hazardous & zm]),
+                    "gas",
+                    "hazardous",
+                    zn,
+                ),
+            ]
+        ]
+    return [
+        c
+        for w in co2_data.data_list
+        for zn, zm in zone_map.items()
+        for c in [
+            ContainedCo2(
+                w.date, sum(w.gas_phase[is_contained & zm]), "gas", "contained", zn
+            ),
+            ContainedCo2(
+                w.date, sum(w.gas_phase[is_outside & zm]), "gas", "outside", zn
+            ),
+            ContainedCo2(
+                w.date, sum(w.gas_phase[is_hazardous & zm]), "gas", "hazardous", zn
+            ),
+            ContainedCo2(
+                w.date, sum(w.aqu_phase[is_contained & zm]), "aqueous", "contained", zn
+            ),
+            ContainedCo2(
+                w.date, sum(w.aqu_phase[is_outside & zm]), "aqueous", "outside", zn
+            ),
+            ContainedCo2(
+                w.date, sum(w.aqu_phase[is_hazardous & zm]), "aqueous", "hazardous", zn
+            ),
+        ]
+    ]
+
+
+def _calculate_containment(
+    x_coord: np.ndarray, y_coord: np.ndarray, poly: Union[Polygon, MultiPolygon]
+) -> np.ndarray:
+    """
+    Determines if (x,y) coordinates belong to a given polygon.
+
+    Args:
+        x_coord (np.ndarray): x coordinates
+        y_coord (np.ndarray): y coordinates
+        poly (Union[Polygon, MultiPolygon]): The polygon that determines the
+                                             containment of the (x,y) coordinates
+
+    Returns:
+        np.ndarray
+    """
+    return np.array([poly.contains(Point(_x, _y)) for _x, _y in zip(x_coord, y_coord)])
+    # try:

```

Comment:
Are these redundant?

Reply:
The code using Pygeos is an alternative code with speed-up, if Pygeos is available. I can remove this code for now.